### PR TITLE
doc: update for new stack macros

### PR DESCRIPTION
**Pull request by tyrantlizard77**
__stack is now deprecated.

Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>

### CODE REVIEW

**Review by nashif — DISMISSED** [review]

File: doc/kernel/threads/lifecycle.rst [hunk 1]
```diff
@@ -237,3 +240,8 @@ The following thread APIs are provided by :file:`kernel.h`:
 * :cpp:func:`k_thread_abort()`
 * :cpp:func:`k_thread_suspend()`
 * :cpp:func:`k_thread_resume()`
+* :c:macro:`K_THREAD_STACK_DEFINE`
+* :c:macro:`K_THREAD_STACK_ARRAY_DEFINE`
+* :c:macro:`K_THREAD_STACK_MEMBERE`
```
Comment (dbkinder) [inline doc/kernel/threads/lifecycle.rst]: MEMBER

**Review by dbkinder — CHANGES_REQUESTED** [review]
one misspelling...

**Comment by andrewboie** [comment]
updated

**Review by dbkinder — APPROVED** [review]
works for me.

**Pull request merged**